lib/fluent/plugin/output.rb

```diff
@@ -326,6 +326,7 @@ def start
           @buffering = prefer_buffered_processing
           if !@buffering && @buffer
             @buffer.terminate # it's not started, so terminate will be enough
+            @buffer = nil
```

Comment:
Non-buffered output plugin (without `#write` method) doesn't have values on `@buffer`.
And, output plugin instances which runs this line is determined as non-buffered plugin. So the `@buffer` value is unassigned.
No bad/wrong things there are.